auth/src/main/java/com/firebase/ui/auth/ui/account_link/WelcomeBackIDPPrompt.java

```diff
@@ -186,7 +194,33 @@ public void onComplete(@NonNull Task<AuthResult> task) {
             authResultTask
                     .addOnFailureListener(
                             new TaskFailureLogger(TAG, "Error linking with credential"))
-                    .addOnCompleteListener(new FinishListener());
+                    .addOnCompleteListener(new OnCompleteListener<AuthResult>() {
+                        @Override
+                        public void onComplete(@NonNull Task<AuthResult> task) {
+                            FirebaseAuth auth = FirebaseAuth.getInstance();
+                            FirebaseUser user = auth.getCurrentUser();
+                            if (!task.isSuccessful() && user != null && mPrevCredential != null) {
+                                user.delete();
```

Comment:
Can you explain exactly what you're doing in this bit?  I'm not clear on what you are trying to achieve.